src/components/ec-metroline/ec-metroline.vue

```diff
@@ -7,62 +7,55 @@
 
-<script>
-export default {
-  name: 'EcMetroline',
-  provide() {
-    return {
-      metroline: this.metroline,
-    };
-  },
-  emits: ['change', 'complete'],
-  data() {
-    return {
-      metroline: {
-        activeItemId: null,
-        lastItemId: null,
-        isCompleted: false,
-        register: (id) => {
-          this.metrolineItemIds.push(id);
-          this.metrolineItemIds.sort();
+<script setup>
+import { provide, reactive, ref } from 'vue';
+
+const emit = defineEmits(['change', 'complete']);
 
-          this.metroline.lastItemId = this.metrolineItemIds[this.metrolineItemIds.length - 1];
+const metroline = reactive({
+  activeItemId: null,
+  lastItemId: null,
+  isCompleted: false,
+  register: (id) => {
+    metrolineItemIds.value.push(id);
+    metrolineItemIds.value.sort();
 
-          if (this.metrolineItemIds.length === 1) {
-            this.metroline.activeItemId = id;
-          }
-        },
-        unregister: (id) => {
-          this.metrolineItemIds = this.metrolineItemIds.filter(item => item !== id);
-        },
-        goToNext: (id) => {
-          if (this.metroline.activeItemId > id) {
-            return;
-          }
+    metroline.lastItemId = metrolineItemIds.value[metrolineItemIds.value.length - 1];
+    if (metrolineItemIds.value.length === 1) {
+      metroline.activeItemId = id;
+    }
+  },
+  unregister: (id) => {
+    metrolineItemIds.value = metrolineItemIds.value.filter(item => item !== id);
+  },
+  goToNext: (id) => {
+    if (metroline.activeItemId > id) {
+      return;
+    }
 
-          const currentIndex = this.metrolineItemIds.findIndex(item => item === id);
-          const nextMetrolineItemId = this.metrolineItemIds[currentIndex + 1];
-          if (nextMetrolineItemId) {
-            this.metroline.activeItemId = nextMetrolineItemId;
-            this.$emit('change', this.metroline.activeItemId);
-          } else {
-            this.metroline.complete();
-          }
-        },
-        goTo: (id) => {
-          this.metroline.activeItemId = id;
-          this.metroline.isCompleted = false;
-          this.$emit('change', this.metroline.activeItemId);
-        },
-        complete: () => {
-          this.metroline.isCompleted = true;
-          this.$emit('complete');
-        },
-      },
-      metrolineItemIds: [],
-    };
+    const currentIndex = metrolineItemIds.value.findIndex(item => item === id);
+    const nextMetrolineItemId = metrolineItemIds.value[currentIndex + 1];
+    if (nextMetrolineItemId) {
+      metroline.activeItemId = nextMetrolineItemId;
+      emit('change', metroline.activeItemId);
+    } else {
+      metroline.complete();
+    }
   },
-};
+  goTo: (id) => {
+    metroline.activeItemId = id;
+    metroline.isCompleted = false;
+    emit('change', metroline.activeItemId);
+  },
+  complete: () => {
+    metroline.isCompleted = true;
+    emit('complete');
+  },
+});
+
+const metrolineItemIds = ref([]);
+
+provide('metroline', metroline);
```

Comment:
while we providing something it is better to use Symbol as identifier, because strings are not unique.

Reply:
Agree, let's create a symbol in this file, export it and import it in metroline-item. I don't think we should be moving it to a separate file.

Reply:
At the end I moved it externally in a folder called "Symbols". Seems script setup does not like exports.